kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/kms/aws/KubeAwsKmsTestKmsFacade.java

```diff
@@ -0,0 +1,234 @@
+            return new AwsKmsLocal();
+        }
+        else {
+            return new AwsKms();
+        }
+    }
+
+    /**
+     * Gets kek key id.
+     *
+     * @return the kek key id
+     */
+    public String getKekKeyId() {
+        return kekKeyId;
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return awsKmsClient.isAvailable();
+    }
+
+    @Override
+    public void startKms() {
+        awsKmsClient.deploy();
+    }
+
+    @Override
+    public void stopKms() {
+        awsKmsClient.delete();
+    }
+
+    @NonNull
+    @Override
+    protected URI getAwsUrl() {
+        return awsKmsClient.getAwsUrl();
+    }
+
+    @Override
+    protected String getRegion() {
+        return awsKmsClient.getRegion();
+    }
+
+    @Override
+    protected String getSecretKey() {
+        return awsKmsClient.getSecretKey();
+    }
+
+    @Override
+    protected String getAccessKey() {
+        return awsKmsClient.getAccessKey();
+    }
+
+    @Override
+    public TestKekManager getTestKekManager() {
+        return new AwsKmsTestKekManager();
+    }
+
+    class AwsKmsTestKekManager implements TestKekManager {
+        @Override
+        public void generateKek(String alias) {
+            Objects.requireNonNull(alias);
+
+            if (exists(alias)) {
+                throw new AlreadyExistsException(alias);
+            }
+            else {
+                create(alias);
+            }
+        }
+
+        @Override
+        public void rotateKek(String alias) {
+            Objects.requireNonNull(alias);
+
+            if (!exists(alias)) {
+                throw new UnknownAliasException(alias);
+            }
+            else {
+                rotate(alias);
+            }
+        }
+
+        @Override
+        public void deleteKek(String alias) {
+            if (!exists(alias)) {
+                throw new UnknownAliasException(alias);
+            }
+            else {
+                delete(alias);
+            }
+        }
+
+        @Override
+        public boolean exists(String alias) {
+            try {
+                read(alias);
+                return true;
+            }
+            catch (KubeClusterException nfe) {
+                return false;
+            }
+        }
+
+        private void create(String alias) {
+            var createKeyResponse = runAwsKmsCommand(CREATE_KEY_RESPONSE_TYPE_REF, awsCmd, KMS, CREATE);
+            kekKeyId = createKeyResponse.keyMetadata().keyId();
+
+            runAwsKmsCommand(awsCmd, KMS, CREATE_ALIAS, PARAM_ALIAS_NAME, ALIAS_PREFIX + alias, PARAM_TARGET_KEY_ID, kekKeyId);
+        }
+
+        private DescribeKeyResponse read(String alias) {
+            return runAwsKmsCommand(DESCRIBE_KEY_RESPONSE_TYPE_REF, awsCmd, KMS, DESCRIBE_KEY, PARAM_KEY_ID, ALIAS_PREFIX + alias);
+        }
+
+        private void rotate(String alias) {
+            // RotateKeyOnDemand is not implemented in localstack.
```

Comment:
see https://github.com/kroxylicious/kroxylicious/issues/1338

Reply:
Should we fix this in the issue you've created? or should I do it here? it is really simple once AwsLocal and AwsCloud are split